File: Java/CodesInterviews/src/main/java/org/example/facebook/MoveZeroes.java
package org.example.facebook;

import java.util.Arrays;

public class MoveZeroes {

    public void moveZeroes(int[] nums) {
        int zeroIndex = 0;
        int n = nums.length;

        for (int nonZeroIndex = 0; nonZeroIndex < n; nonZeroIndex++) {
            if (nums[nonZeroIndex] != 0) {
                nums[zeroIndex] = nums[nonZeroIndex];
                zeroIndex++;
            }
        }

        for (int i = zeroIndex; i < n; i++) {
            nums[i] = 0;
        }

        System.out.println(Arrays.toString(nums));
    }

    public static void main(String[] args) {
        MoveZeroes moveZeroes = new MoveZeroes();
        int[] nums = {0, 1, 0, 3, 12};
        moveZeroes.moveZeroes(nums);
    }
}

File: Kotlin/CodesInterviews/src/main/kotlin/facebook/MoveZeroes.kt
package facebook

import java.util.Arrays

class MoveZeroes {

    fun moveZeroes(nums: Array<Int>) {
        var zeroIndex = 0
        val n = nums.size

        for (nonZeroIndex in 0..<n) {
            if (nums[nonZeroIndex] != 0) {
                nums[zeroIndex] = nums[nonZeroIndex];
                zeroIndex++
            }
        }

        for (i in zeroIndex..<n ) {
            nums[i] = 0
        }

        println(nums.contentToString())
    }

}

fun main() {
    val moveZeroes = MoveZeroes()
    val nums = arrayOf(0, 1, 0, 3, 12)
    moveZeroes.moveZeroes(nums)
}